gitpod-io/gitpod
# [code-browser] extensions observabiilty

## Description


This PR adds RED metrics for queries to the extension gallery and user operations to manage extensions like install and uninstall from user perspective. It allows unambiguously gauge impact of a situation when open-vsx.org is unavailable, i.e. whether our caches are helpful.

[VC Code extensions observability - Watch Video](https://www.loom.com/share/9fa2a3d2e68b44ec9c995c387b30a271)

A supporting PR in openvsscode-server: https://github.com/gitpod-io/openvscode-server/pull/422

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11608

## How to test


- Enable **latest** VS Code Browser : https://akosyakov-f350eda236.preview.gitpod-dev.com/preferences
- Start a prev workspace: https://akosyakov-f350eda236.preview.gitpod-dev.com#https://github.com/akosyakov/parcel-demo, search for VS Code extensions like vim and goland, install and uninstall them.
- Start a dev workspace: https://gitpod.io#https://github.com/gitpod-io/gitpod/pull/12539
- Port forward monitoring: `./dev/preview/portforward-monitoring-satellite.sh -c harvester`
- Open Grafana metrics on port 3000 and have a look at `VS Code Browser Overview` dashboard under IDE Team folder. You can see throughput of queries to the extension gallery, failure rates and latency as its observed by end users. You can also review throughput of user operations like extension installation/uninstallation/upgrade, its failure rates and latencies.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
